src/server/handlers/auth.rs

```diff
@@ -60,30 +65,56 @@ impl<'a, 'b> Claims<'a, 'b> {
     }
 }
 
-pub async fn login(req: Json<model::LoginRequest>) -> Result<HttpResponse, Error> {
-    let user_id = match &req.challenge {
+pub async fn login<T: Store>(
+    req: Json<model::LoginRequest>,
+    storage: Data<T>,
+) -> Result<HttpResponse, Error> {
+    let req = req.into_inner();
+    let user_id = storage
+        .fetch_user_id(&req.identifier)
```

Comment:
Should we move this to model somewhere.  What I typically do is have handlers just pass db connection and request data to models and let model figure out rest.

- Makes it easier to reason about the overall business logic in single place regardless of webserver
- Should make it easy to swap out webservers if needed
- Helps to avoid other devs from adding too much logic to handlers (aka controllers in mvc) and potentially duplicating logic code in different handlers